apps/aemon/src/aemon_mon_on_chain.erl

```diff
@@ -0,0 +1,99 @@
+-module(aemon_mon_on_chain).
+-behaviour(gen_server).
+
+-export([notify/1]).
+-export([start_link/0]).
+-export([ init/1
+        , handle_call/3
+        , handle_cast/2
+        , handle_info/2
+        , terminate/2
+        , code_change/3
+        ]).
+
+notify(Height) ->
+    gen_server:cast(?MODULE, {gen, Height}).
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+%% gen_server callback
+
+init(_) ->
+    aemon_metrics:create(on_chain),
+    PubKey = aemon_config:pubkey(),
+    {ok, PubKey}.
+
+terminate(_Reason, _St) ->
+    ok.
+
+code_change(_FromVsn, St, _Extra) ->
+    {ok, St}.
+
+handle_call(_Req, _From, St) ->
+    {reply, {error, unknown_request}, St}.
+
+handle_cast({gen, Height}, St) ->
+    handle_gen(Height, St),
+    {noreply, St};
+handle_cast(_Msg, St) ->
+    {noreply, St}.
+
+handle_info(_Msg, St) ->
+    {noreply, St}.
+
+%% internals
+
+handle_gen(Height, PubKey) ->
+    try
+        {ok, #{micro_blocks := Blocks}} = aec_chain:get_generation_by_height(Height, forward),
+        [ handle_tx(Height, Tx, aetx_sign:hash(SignTx)) ||
+          Block <- Blocks,
+          SignTx <- aec_blocks:txs(Block),
+          Tx <- [ aetx_sign:tx(SignTx) ],
+          aetx:origin(Tx) == PubKey
+        ],
+        ttl_gen(Height)
+    catch
+        _:_ -> ok
```

Comment:
Would it be possible to be a bit more selective in the cases to handle silently, and log the others? I am concerned that when code refactored / changed then this would require much time to identify - especially as this is in a cast.

Reply:
On it

Reply:
update